lib/ansible/utils/unsafe_proxy.py

```diff
@@ -114,26 +114,67 @@ def _wrap_set(v):
 
 
 def wrap_var(v):
+    """Recursively wraps a variable with ``AnsibleUnsafe`` subclasses to prevent values from being
+    templated by the template engine
+
+    Supports:
+        bytes
+        text
+        Mapping (dict)
+        MutableSequence (list)
+        Set
+    """
     if v is None or isinstance(v, AnsibleUnsafe):
         return v
 
-    if isinstance(v, Mapping):
-        v = _wrap_dict(v)
+    if isinstance(v, binary_type):
+        return AnsibleUnsafeBytes(v)
+    elif isinstance(v, text_type):
+        return AnsibleUnsafeText(v)
     elif isinstance(v, Set):
-        v = _wrap_set(v)
+        return _wrap_set(v)
+    elif isinstance(v, Mapping):
+        return _wrap_dict(v)
     elif is_sequence(v):
-        v = _wrap_sequence(v)
-    elif isinstance(v, binary_type):
-        v = AnsibleUnsafeBytes(v)
-    elif isinstance(v, text_type):
-        v = AnsibleUnsafeText(v)
+        return _wrap_sequence(v)
 
     return v
```

Comment:
Do we still need this return statement here? What types would fall through to this return?

Reply:
`bool`, `int`, `RandomType`...